doc/source/whatsnew/v0.25.0.rst

```diff
@@ -227,7 +227,7 @@ Performance Improvements
 
 Bug Fixes
 ~~~~~~~~~
-
+- Bug in `SparseDataFrame` when adding a column in which the length of values does not match length of index, ``AssertionError`` is raised instead of raising ``ValueError`` (:issue:`25484`)
```

Comment:
can you make a Sparse sub-section and move this there

Reply:
okay, sparse section is already there, moved! thanks @jreback